rts/lra/coordinator/src/test/java/io/narayana/lra/coordinator/domain/model/LRATest.java

```diff
@@ -801,6 +803,42 @@ public void testCancel() {
         runLRA(true);
     }
 
+    /**
+     * This test checks that LRA can be used with a JDBC Object Store
+     */
+    @Test
+    public void jdbcStoreTest() {
+
+        BeanPopulator.getDefaultInstance(ObjectStoreEnvironmentBean.class).setObjectStoreType(JDBCStore.class.getName());
+        String objectStoreType = BeanPopulator.getDefaultInstance(ObjectStoreEnvironmentBean.class).getObjectStoreType();
+
+        LRALogger.logger.infof("The Object Store type is: %s", objectStoreType);
+
+        URI lraIdUri = lraClient.startLRA("ObjectStoreTestCase#newLRA");
+
+        assertNotNull("LRA should have been added to the object store", lraIdUri);
+
+        assertEquals("Expected Active", LRAStatus.Active, getStatus(lraIdUri));
```

Comment:
Would it also be possible to run an SQL query against the JDBC object store to verify that the record made it to the database as expected.

Maybe also try closing the LRA and then use another SQL query to validate that the transaction log has been removed.

Maybe also check that the expected attributes of a LongRunningAction are populated in the result set of an SQL query.

Reply:
Thanks for your review :-) No problem to add the SQL queries.

Just to double check, I thought that once I am sure that the Object Store is set to the desired type (in this case, JDBCStore), the method getStatus(URI uri) of NarayanaLRAClient already verifies the existence of the record for us. So, for example, if I close the LRA and then I try to get its status again, I should get a NotFoundException.

Reply:
Yes, that's the theory but the goal of testing is to verify the hypotheses that the developer got it right. Examples:

- some strange execution path ends up writing the log in a different store type;
- some of the attributes of the record are missing or have the wrong value; 
- the record gets written to the wrong table;
- etch

Reply:
Thanks for your comment @mmusgrov, I'll add some code to programmatically check our hypotheses :+1: